### PR TITLE
[Doc] range partitioning column supports timestamp&string 

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/zh/table_design/Data_distribution.md [hunk 1]
```diff
@@ -197,44 +198,142 @@ StarRocks 支持单独和组合使用数据分布方式。
...
+  AGGREGATE KEY(event_time, site_id, city_code, user_name)
+  PARTITION BY RANGE(from_unixtime(event_time)) (
+      PARTITION p1 VALUES LESS THAN ("2021-01-01"),
+      PARTITION p2 VALUES LESS THAN ("2021-01-02"),
+      PARTITION p3 VALUES LESS THAN ("2021-01-03")
+  )
+  DISTRIBUTED BY HASH(site_id)
+  ;
+  
+  -- 13 位的时间戳类型，例如：1703832553219
```
Comment (jaogoy) [inline docs/zh/table_design/Data_distribution.md:281]: 精确到毫秒的13位。。。

File: docs/zh/table_design/Data_distribution.md [hunk 2]
```diff
@@ -197,44 +198,142 @@ StarRocks 支持单独和组合使用数据分布方式。
...
+
+- 分区列为时间戳和字符串类型（自 3.3.0 起支持）
+  
+  在显式定义的分区与分区列值范围的映射关系时，需要先使用函数将时间戳类型或字符串的分区列值转成日期类型，然后按转换后的日期划分分区。
+
+  <Tabs groupId="manual partitioning">
+  <TabItem value="example1" label="分区列为时间戳" default>
+
+  ```SQL
+  -- 10 位的时间戳类型，例如：1703832553
```
Comment (jaogoy) [inline docs/zh/table_design/Data_distribution.md:264]: 精确到秒的10位。。。

File: docs/zh/table_design/Data_distribution.md [hunk 3]
```diff
@@ -348,15 +442,83 @@ DISTRIBUTED BY HASH(site_id);
...
+  -- 13 位的时间戳类型，例如：1703832553219
+  CREATE TABLE site_access(
+      event_time bigint,
+      site_id INT,
+      city_code SMALLINT,
+      user_name VARCHAR(32),
+      pv BIGINT DEFAULT '0'
+  )
+  PARTITION BY RANGE(from_unixtime_ms(event_time))(
+  START ("2021-01-01") END ("2021-01-10") EVERY (INTERVAL 1 DAY)
```
Comment (jaogoy) [inline docs/zh/table_design/Data_distribution.md:497]: 锁进

File: docs/zh/table_design/Data_distribution.md [hunk 4]
```diff
@@ -348,15 +442,83 @@ DISTRIBUTED BY HASH(site_id);
...
-    ADD PARTITIONS START ("2021-01-04") END ("2021-01-06") EVERY (INTERVAL 1 DAY);
+    CREATE TABLE site_access (
+         event_time  varchar(100),
+         site_id INT,
+         city_code SMALLINT,
+         user_name VARCHAR(32),
+         pv BIGINT DEFAULT '0'
+  )
+    PARTITION BY RANGE(str2date(event_time, '%Y-%m-%d'))(
+    START ("2021-01-01") END ("2021-01-10") EVERY (INTERVAL 1 DAY)
```
Comment (jaogoy) [inline docs/zh/table_design/Data_distribution.md:514]: 缩紧

File: docs/zh/table_design/Data_distribution.md [hunk 5]
```diff
@@ -348,15 +442,83 @@ DISTRIBUTED BY HASH(site_id);
...
+  -- 10 位的时间戳类型，例如：1703832553
+  CREATE TABLE site_access(
+      event_time bigint,
+      site_id INT,
+      city_code SMALLINT,
+      user_name VARCHAR(32),
+      pv BIGINT DEFAULT '0'
+    )
+  PARTITION BY RANGE(from_unixtime(event_time)) (
+  START ("2021-01-01") END ("2021-01-10") EVERY (INTERVAL 1 DAY)
```
Comment (jaogoy) [inline docs/zh/table_design/Data_distribution.md:485]: 缩进

File: docs/zh/table_design/Data_distribution.md [hunk 6]
```diff
@@ -348,15 +442,83 @@ DISTRIBUTED BY HASH(site_id);
...
+        site_id INT,
+        city_code SMALLINT,
+        user_name VARCHAR(32),
+        pv BIGINT DEFAULT '0'
+    )
+    DUPLICATE KEY(datekey, site_id, city_code, user_name)
+    PARTITION BY RANGE (datekey) (
+        START ("1") END ("10") EVERY (1),
+        START ("10") END ("100") EVERY (10)
+
```
Comment (jaogoy) [inline docs/zh/table_design/Data_distribution.md:462]: 多空行

**Review by jaogoy — DISMISSED** [review]

File: docs/en/table_design/Data_distribution.md [hunk 7]
```diff
@@ -199,166 +200,327 @@ You only need to configure a partition expression (a time function expression or
...
 
-[Dynamic partitioning](./dynamic_partitioning.md) related properties are configured at table creation. StarRocks automatically creates new partitions in advance and removes expired partitions to ensure data freshness, which implements time-to-live (TTL) management for partitions.
+:::info
 
-Different from the automatic partition creation ability provided by the expression partitioning, dynamic partitioning can only periodically create new partitions based on the properties. If the new data does not belong to these partitions, an error is returned for the load job. However, the automatic partition creation ability provided by the expression partitioning can always create corresponding new partitions based on the loaded data.
+- If the partitioning column value is a timestamp, you need to use the from_unixtime or from_unixtime_ms function to convert the timestamp to the date when dividing the partitions. When the from_unixtime function is used, the partitioning column only supports INT and BIGINT types. When using the from_unixtime_ms function is used, the partitioning column only supports BIGINT type.
```
Comment (EsoragotoSpirit) [inline docs/en/table_design/Data_distribution.md:211]: the date —> a date value ?

File: docs/en/table_design/Data_distribution.md [hunk 8]
```diff
@@ -199,166 +200,327 @@ You only need to configure a partition expression (a time function expression or
...
+- If the partitioning column value is a string (STRING, VARCHAR, or CHAR types), you need to use the str2date function to convert the string to the date when dividing the partitions.
```
Comment (EsoragotoSpirit) [inline docs/en/table_design/Data_distribution.md:212]: the date —> a date value ?

**Review by EsoragotoSpirit — DISMISSED** [review]

File: docs/en/table_design/Data_distribution.md [hunk 9]
```diff
@@ -199,166 +200,327 @@ You only need to configure a partition expression (a time function expression or
...
 
-StarRocks stores data in the corresponding partitions based on the explicit mapping of the explicitly defined range for each partition.
+You need to explicitly define the data partitioning columns and establish the mapping relationship between partitions and ranges of partitioning column values. During data loading, StarRocks assigns the data to the corresponding partitions based on the ranges to which the data partitioning column values belong.
 
-##### Dynamic partitioning
+As for the data type of the partitioning columns, before v3.3.0, range partitioning only supports partitioning columns of date and integer types. Since v3.3.0, three specific time functions to be used as partition columns. When explicitly defining the mapping relationship between partitions and ranges of partitioning column values, you need to first use a specific time function to convert partitioning column values of timestamps or strings into dates, and then divide the partitions based on the converted dates.
```
Comment (evelynzhaojie) [inline docs/en/table_design/Data_distribution.md:207]: to be used --- 语法不对，can be used

File: docs/en/table_design/Data_distribution.md (hunk 7 above)
Comment (evelynzhaojie) [inline docs/en/table_design/Data_distribution.md:211]: When using the from_unixtime_ms function is used, --- When the from_unixtime_ms function is used,

File: docs/en/table_design/Data_distribution.md [hunk 10]
```diff
@@ -199,166 +200,327 @@ You only need to configure a partition expression (a time function expression or
...
 
-  In the following example, the date range of all the partitions created in batch starts from 2021-01-01 and ends on 2021-01-04, with an incremental interval of one day:
+  <Tabs groupId="manual partitioning">
+  <TabItem value="example1" label="The partition column values are timestamps" default>
 
-    ```SQL
-  CREATE TABLE site_access (
-      datekey DATE,
+  ```SQL
+  -- A 10-digit timestamp accurate to seconds, for example: 1703832553.
```
Comment (evelynzhaojie) [inline docs/en/table_design/Data_distribution.md:266]: A 10-digit timestamp accurate to the second, for example, 1703832553

File: docs/en/table_design/Data_distribution.md [hunk 11]
```diff
@@ -199,166 +200,327 @@ You only need to configure a partition expression (a time function expression or
...
+  PARTITION BY RANGE(from_unixtime(event_time)) (
+      PARTITION p1 VALUES LESS THAN ("2021-01-01"),
+      PARTITION p2 VALUES LESS THAN ("2021-01-02"),
+      PARTITION p3 VALUES LESS THAN ("2021-01-03")
   )
   DISTRIBUTED BY HASH(site_id)
-  PROPERTIES ("replication_num" = "3" );
+  ;
+  
+  -- A 13-digit timestamp accurate to milliseconds, for example: 1703832553219.
```
Comment (evelynzhaojie) [inline docs/en/table_design/Data_distribution.md:283]: A 13-digit timestamp accurate to the millisecond, for example, 1703832553219

File: docs/en/table_design/Data_distribution.md [hunk 12]
```diff
@@ -199,166 +200,327 @@ You only need to configure a partition expression (a time function expression or
...
+
+Different from the automatic partition creation ability provided by the expression partitioning, dynamic partitioning can only periodically create new partitions based on the properties. If the new data does not belong to these partitions, an error is returned for the load job. However, the automatic partition creation ability provided by the expression partitioning can always create corresponding new partitions based on the loaded data.
+
+##### Create multiple partitions in batch
+
+Multiple partitions can be created in batch at and after table creation. You can specify the start and end time for all the partitions created in batch in `START()` and `END()` and the partition increment value in `EVERY()`. However, note that the range of partitions is right hand half open, which includes the start time but does not include the end time. The naming rule for partitions is the same as that of dynamic partitioning.
```
Comment (evelynzhaojie) [inline docs/en/table_design/Data_distribution.md:331]: right hand half open -- left-closed and right-open?

File: docs/en/table_design/Data_distribution.md [hunk 13]
```diff
@@ -199,166 +200,327 @@ You only need to configure a partition expression (a time function expression or
...
+
+- **The partitioning column is of date type.**
+
+  When the partitioning column is of date type, at table creation, you can use `START()` and `END()` to specify the start date and end date for all the partitions created in batch, and `EVERY(INTERVAL xxx)` to specify the incremental interval between two partitions. Currently the interval granularity supports `HOUR` (since v3.0), `DAY`, `WEEK`, `MONTH`, and `YEAR`.
+
+  <Tabs groupId="batch partitioning(date)">
+  <TabItem value="example1" label="with the same date interval" default>
+  
+  In the following example, the partitions created in a batch start from `2021-01-01` and ends on `2021-01-04`, with a partition increment of one day:
```
Comment (evelynzhaojie) [inline docs/en/table_design/Data_distribution.md:340]: and ends on -- and end on

File: docs/en/table_design/Data_distribution.md [hunk 14]
```diff
@@ -199,166 +200,327 @@ You only need to configure a partition expression (a time function expression or
...
+
+- **Three specific time functions can be used as partitioning columns (supported since v3.3.0).**
+  
+  When explicitly defining the mapping relationship between partitions and the ranges of partition column values, you can use a specific time function to convert the partition column values of timestamps or strings into dates, and then divide the partitions based on the converted dates.
+
+  <Tabs groupId="batch partitioning(timestamp and string)">
+  <TabItem value="example1" label="The partitioning column values are timestamps" default>
+
+  ```SQL
+  -- A 10-digit timestamp accurate to seconds, for example: 1703832553.
```
Comment (evelynzhaojie) [inline docs/en/table_design/Data_distribution.md:479]: A 10-digit timestamp accurate to the second, for example, 1703832553

File: docs/en/table_design/Data_distribution.md [hunk 15]
```diff
@@ -199,166 +200,327 @@ You only need to configure a partition expression (a time function expression or
...
+      site_id INT,
+      city_code SMALLINT,
+      user_name VARCHAR(32),
+      pv BIGINT DEFAULT '0'
+    )
+  PARTITION BY RANGE(from_unixtime(event_time)) (
+      START ("2021-01-01") END ("2021-01-10") EVERY (INTERVAL 1 DAY)
+  )
+  DISTRIBUTED BY HASH(site_id);
+  -- A 13-digit timestamp accurate to milliseconds, for example: 1703832553219.
```
Comment (evelynzhaojie) [inline docs/en/table_design/Data_distribution.md:491]: A 13-digit timestamp accurate to the milliseconds, for example, 1703832553219

**Review by evelynzhaojie — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Pull request merged**